test/types/records/engin/comparison.chpl

```diff
@@ -60,15 +58,17 @@ write("Class field with operators overloaded... ");
   var small2 = new R(1);
   var big = new R(2);
 
-  test(small, small2, big);
+  test(small, small2, big, true);
 }
 writeln("success");
 
-proc test(small, small2, big) {
-  assert((small == small2) == true);
-  assert((small != small2) == false);
-  assert((small == big) == false);
-  assert((small != big) == true);
+proc test(small, small2, big, skipequals=false) {
+  if (!skipequals) {
+    assert((small == small2) == true);
+    assert((small != small2) == false);
+    assert((small == big) == false);
+    assert((small != big) == true);
```

Comment:
This makes me wonder, though.. Shouldn't I still be able to `==` on two identical-typed records that contain class fields? Where `==` on the field would be just a nil check?

Reply:
By default, `==` on the class fields will ask "do they point to the same object?"  For your records, they don't, so the assertions started failing once I removed the user-defined overloads.

Reply:
Looking at https://github.com/chapel-lang/chapel/pull/16824 more: This test still looks like it should be legal to me, based on the fact that `==` is well-defined on classes, and records should be able to make use of that for their fields. In other words, we are only barring users from being able to overload its meaning, they can still use it either directly or indirectly (as this test does)

Reply:
Right, but trying again:  as currently defined, `small` and `small2` each point to different a class object so while `==` and `!=` can be applied to them, the wrong answers result (i.e., `==` is `false` and `!=` is `true`) so the assertions fail and the test fails.  Here, I worked around that by skipping the failing assertions in this case.  Another approach would be to have the records point to the same class (but then the `<` and friends assertions would fail).  Another approach would be to have an `else` in this conditional that reverses the expected outcomes.

Reply:
Oh OK. (I hadn't seen your first response when I posted my second one, sorry for the confusion)

Reply:
Got it.  I'm happy to make one of those other changes if you prefer.

Reply:
I don't have a strong preference.

Reply:
OK, I did a refactor that I'm pretty happy with if you want to look again.

Reply:
Nice :)